app/src/main/java/tech/ula/utils/AndroidUtility.kt

```diff
@@ -398,4 +399,54 @@ class AcraWrapper {
     fun putCustomString(key: String, value: String) {
         ACRA.getErrorReporter().putCustomData(key, value)
     }
+}
+
+class UsageUtility(private val prefs: SharedPreferences) {
+    private val numberOfTimesOpenedKey = "numberOfTimesOpened"
+    private val userGaveFeedbackKey = "userGaveFeedback"
+    private val dateTimeFirstOpenKey = "dateTimeFirstOpen"
+    private val millisecondsInThreeDays = 259200000L
+    private val minimumNumberOfOpensBeforeReviewRequest = 15
+
+    fun addToNumberOfTimesOpened(): Int {
+        val numberTimesOpened = prefs.getInt(numberOfTimesOpenedKey, 1)
+
+        if (numberTimesOpened == 1) {
+            with(prefs.edit()) {
+                putInt(numberOfTimesOpenedKey, numberTimesOpened + 1)
+                putLong(dateTimeFirstOpenKey, Date().time)
```

Comment:
You can use `System.currentTimeMillis` to avoid potential issues with `Date` and more closely match what we do elsewhere.

Reply:
I've changed it to use `System.currentTimeMillis` instead of using `Date`